app/ui/lib/Combobox.tsx

```diff
@@ -189,17 +191,25 @@ export const Combobox = ({
           >
             <ComboboxInput
               id={`${id}-input`}
-              // displayValue controls what's displayed in the input field.
-              // selectedItemValue is displayed when the user can type in a new value.
-              // Otherwise, use the provided selectedItemLabel
-              displayValue={() =>
-                allowArbitraryValues ? selectedItemValue : selectedItemLabel
+              // If an option has been selected, display either the selected item's label or value.
+              // If no option has been selected yet, or the user has started editing the input, display the query.
+              // We are using value here, as opposed to Headless UI's displayValue, so we can normalize
+              // the value entered into the input (via the onChange event).
+              value={
+                selectedItemValue
+                  ? allowArbitraryValues
+                    ? selectedItemValue
+                    : selectedItemLabel
+                  : query
               }
               onChange={(event) => {
+                // Pass in true for the second prop so that the user can start their filtering query with a number.
+                // A validation will still prevent them from submitting the form with a number at the beginning.
+                const value = normalizeName(event.target.value, true)
```

Comment:
Maybe this function shouldn't be hard-coded in here. It's true that currently the values are all names, but this is meant to be a generic `Combobox`, not a `NameCombobox`. How about a `transform` prop like `TextField`?

https://github.com/oxidecomputer/console/blob/f3d38103253dab2d2670725a10a7d91b26e7937f/app/components/form/fields/TextField.tsx#L49-L50

https://github.com/oxidecomputer/console/blob/f3d38103253dab2d2670725a10a7d91b26e7937f/app/components/form/fields/TextField.tsx#L121